lib/graphql_metrics/tracer.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module GraphQLMetrics
+  class Tracer
+    # NOTE: Used to store timings from lexing, parsing, validation, before we have a context to store
+    # values in. Uses thread-safe Concurrent::ThreadLocalVar to store a set of values per thread.
+    cattr_accessor :pre_context
+
+    # NOTE: These constants come from the graphql ruby gem.
+    GRAPHQL_GEM_LEXING_KEY = 'lex'
+    GRAPHQL_GEM_PARSING_KEY = 'parse'
+    GRAPHQL_GEM_VALIDATION_KEYS = ['validate', 'analyze_query', 'analyze_multiplex']
+    GRAPHQL_GEM_TRACING_FIELD_KEYS = [
+      GRAPHQL_GEM_TRACING_FIELD_KEY = 'execute_field',
+      GRAPHQL_GEM_TRACING_LAZY_FIELD_KEY = 'execute_field_lazy'
+    ]
+
+    def self.trace(key, data, &resolver_block)
+      # NOTE: Context doesn't exist yet during lexing, parsing.
+      possible_context = data[:query]&.context
+
+      skip_tracing = possible_context&.fetch(GraphQLMetrics::SKIP_GRAPHQL_METRICS_ANALYSIS, false)
+      return yield if skip_tracing
+
+      return setup_tracing_before_lexing { yield } if key == GRAPHQL_GEM_LEXING_KEY
```

Comment:
🤔 this works? I wasn't really clear but you can still do `setup_tracing_before_lexing(&resolver_block)`. Just need `&`

Reply:
Yeah, I'm just yielding all the way through now without any & referenced blocks. I think it still reads fairly nicely.